commitserver/commit/hydratorhelper.go

```diff
@@ -196,6 +216,63 @@ func writeManifests(root *os.Root, dirPath string, manifests []*apiclient.Hydrat
 			return fmt.Errorf("failed to encode manifest: %w", err)
 		}
 	}
+	return nil
+}
 
+// deleteManifest attempts to remove the "manifest.yaml" file located in the specified directory path
+// within the given os.Root abstraction. If the file removal fails, it returns a wrapped error with context.
+// This helper is used to clean up files which are written to disk but are still un-staged and a similar file exists in index.
+func deleteManifest(root *os.Root, dirPath string) error {
+	manifestPath := filepath.Join(dirPath, ManifestYaml)
+	if err := root.Remove(manifestPath); err != nil {
+		return fmt.Errorf("failed to remove manifest: %w", err)
+	}
+	return nil
+}
+
+// hasManifestChanged checks whether the "manifest.yaml" file in the specified directory path
+// has been modified in the working tree compared to the version tracked in the current git index.
+// Returns true if the file has changes staged or unstaged; otherwise, returns false.
+// Relies on the provided git.Client's HasFileChanged method for underlying diff detection.
+func hasManifestChanged(dirPath string, gitClient git.Client) (bool, error) {
+	manifestPath := filepath.Join(dirPath, ManifestYaml)
+	return gitClient.HasFileChanged(manifestPath)
+}
```

Comment:
This seems too trivial to justify a helper function. Can you just inline `gitClient.HasFileChanged(filepath.Join(dirPath, ManifestYaml))` where this is called?